src/tests/simian/mac/admin/groups_test.py

```diff
@@ -0,0 +1,257 @@
+#!/usr/bin/env python
+#
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""groups module tests."""
+
+import httplib
+import logging
+
+import mock
+
+from django.conf import settings
+settings.configure()
+from google.apputils import app
+from google.apputils import basetest
+import tests.appenginesdk
+from simian.mac.admin import groups
+from simian.mac.admin import xsrf
+from tests.simian.mac.common import test
+
+
+@mock.patch.object(xsrf, 'XsrfTokenValidate', return_value=True)
+class AdminGroupsTest(test.RequestHandlerTest):
+
+  def setUp(self):
+    super(AdminGroupsTest, self).setUp()
+    self.group1 = groups.models.base.Group(key_name='test group',
+                                           owners=['user1', 'user4'])
+    self.group2 = groups.models.base.Group(key_name='test group2',
+                                           owners=['user1', 'user2'])
+    self.groups_list = [self.group2, self.group1]
+
+  def GetTestClassInstance(self):
+    request = None
+    response = mock.MagicMock()
+    return groups.Groups(request, response)
+
+  def GetTestClassModule(self):
+    return groups
+
+  def testGet(self, _):
+    """Test get()."""
+    self.mox.StubOutWithMock(self.c, 'IsAdminUser')
```

Comment:
mox is deprecated and should not be used in new tests.

New tests should use testbed and mock.
good example acl_groups_test.py